src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    public Ball getRandomBall() {
+        final int maxRandomNumber = 100;
+        int randomNumber = new Random().nextInt(maxRandomNumber);
+        Colors color = new ColorSupplier().getRandomColor();
```

Comment:
Similar to the Random instance, you should not create a new instance of ColorSupplier every time you need a random color. Instead, instantiate ColorSupplier once as a class-level field and reuse it. This is to avoid unnecessary object creation as mentioned in the checklist.